socket.c

```diff
@@ -249,18 +249,29 @@ static struct data *construct(const char *spec, int mode) {
   if (data) {
     data->sock = -1;
     data->client_sock = -1;
-
     data->mode = mode;
+
+    char *bracket = strchr(spec, ']');
     char *colon = strchr(spec, ':');
+    char *tail;
+
+    if (bracket) {
+      spec++;
```

Comment:
You cant simply ignore the first character, you must check that it is `[` and nothing else.